Issue 614 piece view options

DSCaskey opened the pull request:
Adds new options to toggle pattern piece elements.

- Adds items to the view menu and view tool bar to toggle the seam allowance.
- Adds a context menu to pattern pieces in piece mode which allows the user to toggle piece elements without having to open the tool dialog. 
- Renames "Work Piece Tool" to "Pattern Piece Tool." 
- Changes the term "main path" to "seam line."

closes issue #614

DSCaskey committed:
feat: add new view options  for pattern piece elements

Adds new menu, toolbar & context menu options to toggle view of pattern piece elements.

DSCaskey committed:
Merge branch 'develop' into issue-614-piece-view-options

DSCaskey committed:
update shortcuts

update shortcuts help dialog

slspencer reviewed (commented):
Hey @Douglas, The public.key for kde neon that we add to our linux build to fix some icon problems no longer exists, so our Linux build was failing. I found a fix, not really well documented & many people are complaining about this. SO... I was waiting to merge this pull request until this was fixed. EXCEPT that there's now a qt error with Qt::dec due to the appimage was building with 5.12.2 and I forced it to be 5.15.2, which now generates an error. 
I'll do this PR, but there's going to be errors not attributable to you.

slspencer commented on src/libs/vlayout/vlayoutpiece.cpp:
```diff
@@ -420,7 +420,7 @@ VLayoutPiece VLayoutPiece::Create(const VPiece &piece, const VContainer *pattern
     det.SetMy(piece.GetMy());
 
-    det.SetCountourPoints(piece.MainPathPoints(pattern), piece.IsHideMainPath());
+    det.SetCountourPoints(piece.MainPathPoints(pattern), piece.isHideSeamLine());
```
Thanks for the attention to readability

DSCaskey commented:
Hey Susan... did some checking.

Qt::dec is the same as calling setIntegerBase(10). Also as per the docs:

"By default, when reading numbers from a stream of text, QTextStream will
automatically detect the number's base representation. For example, if the
number starts with "0x", it is assumed to be in hexadecimal form. If it
starts with the digits 1-9, it is assumed to be in decimal form, and so on.
You can set the integer base, thereby disabling the automatic detection, by
calling setIntegerBase
<https://doc.qt.io/qt-6/qtextstream.html#setIntegerBase>()."

It would seem the qDebug() << Qt::dec is redundant (and not neccessary) as
the QTextStream should automatically be detecting base 10. If you ask me...
Like I said I don't think this point will ever be reached -  I'd just
comment the line out, where it could be uncommented should someone make an
change to the parser base causing a bug that needs debugging.

On Fri, Jul 15, 2022, 12:13 AM Douglas Caskey ***@***.***> wrote:

> Yeah... curious - I thought we squashed that deprecation before - guess
> not? At any rate I initially thought we could just delete the line, but
> looking at the full code, it's part of a "stack dump" routine. Which itself
> at this point will probably never be called... and  given that it was in
> the qmu math parser that RT ported, it's probably never been called... it's
> just there - more bloat.
>
> Actually as an aside... since the qmuParser is a port  to Qt C++... I have
> no clue why RT made it a dynamic lib instead of static? It is being built
> along with the apps, so it makes no sense to be dynamic. You build a
> dynamic lib so it can be distributed and others can just link the lib
> without building it - while keeping the source code closed.  It's just an
> added step to copy the built lib from the qmuparser/bin to the Seamly2D/bin
> whether building in Creator or via a Github workflow. ??
>
> On Thu, Jul 14, 2022, 9:46 PM Susan Spencer ***@***.***>
> wrote:
>
>> ***@***.**** approved this pull request.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/FashionFreedom/Seamly2D/pull/615#pullrequestreview-1039695896>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AHTXADWOXLTHHMS3DNZ425DVUC7HRANCNFSM5ZYXGG4Q>
>> .
>> You are receiving this because you were assigned.Message ID:
>> ***@***.***>
>>
>

DSCaskey commented:
Yeah... curious - I thought we squashed that deprecation before - guess
not? At any rate I initially thought we could just delete the line, but
looking at the full code, it's part of a "stack dump" routine. Which itself
at this point will probably never be called... and  given that it was in
the qmu math parser that RT ported, it's probably never been called... it's
just there - more bloat.

Actually as an aside... since the qmuParser is a port  to Qt C++... I have
no clue why RT made it a dynamic lib instead of static? It is being built
along with the apps, so it makes no sense to be dynamic. You build a
dynamic lib so it can be distributed and others can just link the lib
without building it - while keeping the source code closed.  It's just an
added step to copy the built lib from the qmuparser/bin to the Seamly2D/bin
whether building in Creator or via a Github workflow. ??

On Thu, Jul 14, 2022, 9:46 PM Susan Spencer ***@***.***>
wrote:

> ***@***.**** approved this pull request.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/FashionFreedom/Seamly2D/pull/615#pullrequestreview-1039695896>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHTXADWOXLTHHMS3DNZ425DVUC7HRANCNFSM5ZYXGG4Q>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
>
